src/inmanta/config.py

```diff
@@ -134,7 +134,7 @@ def validate_option_request(cls, section, name, default_value):
             # raise Exception("Config name %s not defined in section %s" % (name, section))
             return
         opt = cls.__config_definition[section][name]
-        if not opt.get_default_value() == opt.get_default_value():
+        if opt.get_default_value() is not None and opt.get_default_value() != default_value:
```

Comment:
I think it has to be 

```
if default_value is not None and opt.get_default_value() != default_value:
```

The intent of this check is that all default values passed via the get method are consistent with the declared default value (that is used in the docs, opt.get_default_value()) 

In fact, passing a default value in the get call should be phased out at some point in the future, as it makes generating documentation very difficult. 

looking at the remainder of this class, get_boolean is not consistent with get, `get` automatically performs type conversion and validation based on the declared option config, while `get_boolean` relies on the underlying config parser.